path_oidc_test.go

```diff
@@ -361,8 +373,9 @@ func TestOIDC_Callback(t *testing.T) {
 			},
 		}
 		auth := resp.Auth
-		if diff := deep.Equal(auth, expected); diff != nil {
-			t.Fatal(diff)
+
+		if !reflect.DeepEqual(auth, expected) {
```

Comment:
You find this more readable?

Reply:
When comparing objects with *sockaddr.SockAddrMarshaler, deep will always return a diff.  I dug into this quite a while ago and don’t remember the exact details, but it has to do with comparing pointers versus pointed-to values, and how we are embeddding SockAddr. I just know DeepEqual does the comparison correctly and started using it in these rare cases.

Reply:
Ah, that makes sense.  I'll have to try go-cmp at some point and see if it does any better.